scripts/markdown-code-checker.py

```diff
@@ -78,34 +78,39 @@ def check_output():
     pass
 
 
-def replace():
-    pass
+def replace(path, i, language, first_line, last_line):
+    file_name = f"{path}.snippet-{i}.{language}"
+
+    try:
+        with open(file_name, "r") as f:
+            pretty_content = f.read()
+    except:
+        print(
+            f"[error] Couldn't find the file '{file_name}'. Run --extract to extract the inline code."
+        )
+        return
+
+    with open(path, "r") as f:
+        markdown_content = f.read()
+        lines = markdown_content.split("\n")
+
+    lines[first_line + 1 : last_line - 1] = pretty_content.split("\n")
+
+    with open(path, "w") as f:
+        f.write("\n".join(lines))
```

Comment:
All the 3 times you call `open()` an exception can be raised. And the most likely exceptions are file not found, or access denied.

I suggest putting the whole function (all 3 opens) into one big `try...except`, and then `excepting` the specific exceptions like `FileNotFoundError` for nicer error messages.

You can also add the more generic [`IOError`](https://stackoverflow.com/a/70538223) to cover other ways which reading and writing files can fail. Avoid `except` without an exception type, since it can hide other problems we would want to fix, like `AssertionError`, `AttributeError`, etc.